### PR TITLE
Fix a bug where calling a task with custom strategies would fail beca…

### PR DESCRIPTION
**Pull request by jeyrschabu**
…use of casting error

- Added a guard to ensure the parent task is a pipeline before attempting to cast to a Pipeline

### CODE REVIEW

**Comment by jeyrschabu** [comment]
@ajordens PTAL.

This is the PR for master

**Comment by jeyrschabu** [comment]
@robfletcher PTAL

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy [hunk 1]
```diff
@@ -133,7 +133,7 @@ trait DeploymentDetailsAware {
     if (execution instanceof Pipeline) {
       if (execution.trigger.parentExecution instanceof Pipeline) {
         return execution.trigger.parentExecution
-      } else if (execution.trigger.parentExecution) {
+      } else if (execution.trigger?.isPipeline) {
```
Comment (robfletcher) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: What happens if it _isn't_ a pipeline?

Comment (jeyrschabu) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: > In this situation, the task execution can't have a parent pipeline.

**Pull request merged**